testing/go/types_test.go

```diff
@@ -575,6 +575,70 @@ var typesTests = []ScriptTest{
 			},
 		},
 	},
+	{
+		Name: "Name type",
+		SetUpScript: []string{
+			"CREATE TABLE t_name (id INTEGER primary key, v1 NAME);",
+			"INSERT INTO t_name VALUES (1, 'abcdefghij'), (2, 'klmnopqrst');",
+		},
+		Assertions: []ScriptTestAssertion{
+			{
+				Query: "SELECT * FROM t_name ORDER BY id;",
+				Expected: []sql.Row{
+					{1, "abcdefghij"},
+					{2, "klmnopqrst"},
+				},
+			},
+			{
+				Query: "SELECT * FROM t_name ORDER BY v1 DESC;",
+				Expected: []sql.Row{
+					{2, "klmnopqrst"},
+					{1, "abcdefghij"},
+				},
+			},
+			{
+				Query: "SELECT v1::char(1) FROM t_name WHERE v1='klmnopqrst';",
+				Expected: []sql.Row{
+					{"k"},
+				},
+			},
+			{
+				Query:    "UPDATE t_name SET v1='tuvwxyz' WHERE id=2;",
+				Expected: []sql.Row{},
+			},
+			{
+				Query:    "DELETE FROM t_name WHERE v1='abcdefghij';",
+				Expected: []sql.Row{},
+			},
+			{
+				Query: "SELECT * FROM t_name ORDER BY id;",
```

Comment:
Since we're here, we can add implicit and explicit casts as well.

Reply:
We should also test the length limit as well to ensure we're doing the correct behavior.